cmd/util/handler/logging.go

```diff
@@ -7,70 +7,31 @@ import (
 	"os/signal"
 	"time"
 
-	"github.com/bacalhau-project/bacalhau/pkg/bacerrors"
-	"github.com/bacalhau-project/bacalhau/pkg/util/templates"
 	"github.com/gorilla/websocket"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
-	"k8s.io/kubectl/pkg/util/i18n"
-)
-
-var (
-	logsShortDesc = templates.LongDesc(i18n.T(`
-		Follow logs from a currently executing job
-`))
-
-	//nolint:lll // Documentation
-	logsExample = templates.Examples(i18n.T(`
-		# Follow logs for a previously submitted job
-		bacalhau logs 51225160-807e-48b8-88c9-28311c7899e1
 
-		# Follow output with a short ID 
-		bacalhau logs ebd9bf2f
-`))
+	"github.com/bacalhau-project/bacalhau/pkg/bacerrors"
+	"github.com/bacalhau-project/bacalhau/pkg/logger"
 )
 
-type Msg struct {
-	Tag  uint8
-	Data string
-}
-
-type LogCommandOptions struct {
-	Follow      bool
-	WithHistory bool
-}
-
-func newLogsCmd() *cobra.Command {
-	options := LogCommandOptions{}
-
-	logsCmd := &cobra.Command{
-		Use:     "logs [id]",
-		Short:   logsShortDesc,
-		Example: logsExample,
-		Args:    cobra.ExactArgs(1),
-		PreRun:  applyPorcelainLogLevel,
-		RunE: func(cmd *cobra.Command, cmdArgs []string) error {
-			return logs(cmd, cmdArgs, options)
-		},
-	}
-
-	logsCmd.PersistentFlags().BoolVarP(
-		&options.Follow, "follow", "f", false,
-		`Follow the logs in real-time after retrieving the current logs.`,
-	)
+var LoggingMode = logger.LogModeDefault
 
-	return logsCmd
-}
+const (
+	JSONFormat string = "json"
+	YAMLFormat string = "yaml"
+)
 
-func logs(cmd *cobra.Command, cmdArgs []string, options LogCommandOptions) error {
+func Logs(cmd *cobra.Command, jobID string, follow, history bool) error {
```

Comment:
Even though this is the word "logs", `util` is not the right place for this – this is a stand-alone command for collecting Bacalhau logs, not utils.

Reply:
While it is a command, it's also used to print the execution of a job (via `PrintJobExecution`) if the user passes the `--follow` flag to `create`, `docker run`, or `wasm run`: https://github.com/bacalhau-project/bacalhau/pull/2560/files#diff-34e9172ae6c1602e3ad866de40c1ab94e28de0040e0ec71fbc9effcab8ff47b1R93 (printer.go line 93)

Reply:
Those flags are implemented by passing control to the top level `logs` command :D

Reply:
this is confusing to me as well. Does it make sense to have this part of `logs` command, and have `create`, `docker run` and `wasm run` depend on `logs`? Otherwise, a comment on why this method exists here would be helpful

Reply:
What would make this less confusing? This is the dependency graph:
```
 +-----+    +-----+    +-----+
 |docker|    |wasm|    |create|
 +-----+    +-----+    +-----+
      \       |       /
       \      |      /
              ↓
        +-----------------+
        |PrintJobExecution|
        +-----------------+
              ↓
        +-----+        +-----+
        | log |  <---- | Log |
        +-----+        +-----+
```